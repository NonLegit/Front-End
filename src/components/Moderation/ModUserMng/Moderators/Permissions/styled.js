import {
  FormHelperText, Box, FormControlLabel, styled,
} from '@mui/material';

export const StyledFormControlLabel = styled(FormControlLabel)(() => ({
  padding: '0px 0px 0px 16px',
  margin: '0px 0px 0px 0px',
  '& .MuiTypography-root': { fontSize: '12px', fontWeight: '700' },
  '& .MuiButtonBase-root': { padding: '0px 8px 0px 0px' },
  marginTop: '16px',
}));

export const StyledFormHelperText = styled(FormHelperText)(() => ({
  paddingLeft: '45px',
}));

export const StyledPermissionContainer = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'column',
  ml: 3,
}));
