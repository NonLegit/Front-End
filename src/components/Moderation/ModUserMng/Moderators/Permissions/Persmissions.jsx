import * as React from 'react';
import { Checkbox, Divider } from '@mui/material';
import { StyledFormHelperText, StyledFormControlLabel, StyledPermissionContainer } from './styled';

/**
 * permissions that tha moderator has
 * @component
 * @return {React.Component} - permissions that the moderator has
 */

export default function IndeterminateCheckbox() {
  const [checkedAccess, setChecked1] = React.useState(true);
  const [checkedConfig, setChecked2] = React.useState(true);
  const [checkedFlair, setChecked3] = React.useState(true);
  const [checkedPosts, setChecked4] = React.useState(true);

  const handleChangeAccess = (event) => {
    setChecked1(event.target.checked);
  };
  const handleChangeConfig = (event) => {
    setChecked2(event.target.checked);
  };
  const handleChangeFlairs = (event) => {
    setChecked3(event.target.checked);
  };
  const handleChangePosts = (event) => {
    setChecked4(event.target.checked);
  };

  const handleChangeAll = (event) => {
    handleChangeAccess(event);
    handleChangeConfig(event);
    handleChangeFlairs(event);
    handleChangePosts(event);
  };

  const children = (
    <StyledPermissionContainer>
      <StyledFormControlLabel
        label="Manage Users"
        control={(
          <Checkbox
            id="manage_users"
            disableTouchRipple
            disableRipple
            size="small"
            checked={checkedAccess}
            onChange={handleChangeAccess}
          />
)}
      />
      <StyledFormHelperText>
        Access mod notes, ban and mute users, and approve submitters*.
      </StyledFormHelperText>
      <StyledFormControlLabel
        label="Manage Settings"
        control={(
          <Checkbox
            id="manage_settings"
            disableTouchRipple
            disableRipple
            size="small"
            checked={checkedConfig}
            onChange={handleChangeConfig}
          />
)}
      />
      <StyledFormHelperText>
        Manage community settings, appearance, emojis, rules, and AutoMod*.
      </StyledFormHelperText>
      <StyledFormControlLabel
        label="Manage Flair"
        control={(
          <Checkbox
            id="manage_flair"
            disableTouchRipple
            disableRipple
            size="small"
            checked={checkedFlair}
            onChange={handleChangeFlairs}
          />
)}
      />
      <StyledFormHelperText>
        Create and manage user and post flair.
      </StyledFormHelperText>
      <StyledFormControlLabel
        label="Manage Posts & Comments"
        control={(
          <Checkbox
            id="manage_post"
            disableTouchRipple
            disableRipple
            size="small"
            checked={checkedPosts}
            onChange={handleChangePosts}
          />
)}
      />
      <StyledFormHelperText>
        Access queues, take action on content, and manage collections and events.
      </StyledFormHelperText>
    </StyledPermissionContainer>
  );

  return (
    <>
      <StyledFormControlLabel
        label="Everything"
        control={(
          <Checkbox
            id="everything"
            size="small"
            checked={checkedAccess && checkedConfig && checkedFlair && checkedPosts}
            onChange={handleChangeAll}
          />
        )}
      />
      <StyledFormHelperText>
        Full access including the ability to manage moderator access and permissions.
      </StyledFormHelperText>
      <Divider sx={{ marginTop: '16px' }} />
      {children}
    </>
  );
}
