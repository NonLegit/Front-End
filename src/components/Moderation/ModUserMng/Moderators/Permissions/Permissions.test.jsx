import renderer from 'react-test-renderer';
import Permissions from './Persmissions';

test('test snapshot', async () => {
  const tree = renderer.create(<Permissions />).toJSON();
  expect(tree).toMatchSnapshot();
});
