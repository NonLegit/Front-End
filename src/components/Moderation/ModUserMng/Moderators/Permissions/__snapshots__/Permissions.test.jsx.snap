// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
Array [
  <label
    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vwuj74-MuiFormControlLabel-root"
  >
    <span
      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <input
        checked={true}
        className="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate={false}
        id="everything"
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="CheckBoxIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
    >
      Everything
    </span>
  </label>,
  <p
    className="MuiFormHelperText-root css-18km2ni-MuiFormHelperText-root"
  >
    Full access including the ability to manage moderator access and permissions.
  </p>,
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-uok36i-MuiDivider-root"
  />,
  <div
    className="MuiBox-root css-13snqcs"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vwuj74-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-355xbt-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={true}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          id="manage_users"
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Manage Users
      </span>
    </label>
    <p
      className="MuiFormHelperText-root css-18km2ni-MuiFormHelperText-root"
    >
      Access mod notes, ban and mute users, and approve submitters*.
    </p>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vwuj74-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-355xbt-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={true}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          id="manage_settings"
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Manage Settings
      </span>
    </label>
    <p
      className="MuiFormHelperText-root css-18km2ni-MuiFormHelperText-root"
    >
      Manage community settings, appearance, emojis, rules, and AutoMod*.
    </p>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vwuj74-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-355xbt-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={true}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          id="manage_flair"
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Manage Flair
      </span>
    </label>
    <p
      className="MuiFormHelperText-root css-18km2ni-MuiFormHelperText-root"
    >
      Create and manage user and post flair.
    </p>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vwuj74-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-355xbt-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={true}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          id="manage_post"
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Manage Posts & Comments
      </span>
    </label>
    <p
      className="MuiFormHelperText-root css-18km2ni-MuiFormHelperText-root"
    >
      Access queues, take action on content, and manage collections and events.
    </p>
  </div>,
]
`;
